### PR TITLE
feat: remove resultaat when reopening zaak

### PR DESCRIPTION
**Pull request by felixcicatt**
Make sure the Resultaat that belongs to a Zaak gets removed after re-opening said Zaak
Solves PZ-1675

### CODE REVIEW

File: src/itest/kotlin/nl/lifely/zac/itest/ZakenRESTServiceTest.kt [hunk 1]
```diff
@@ -348,4 +349,105 @@ class ZakenRESTServiceTest : BehaviorSpec({
...
+                requestBodyAsString = """
+            {"reden":"vxcvxcv"}
+                """.trimIndent()
+            )
+
+            KeycloakClient.authenticate()
+
+            val heropenBody = heropenResult.body!!.string()
+
+            logger.info { heropenResult }
```
Comment (edgarvonk) [inline src/itest/kotlin/nl/lifely/zac/itest/ZakenRESTServiceTest.kt:437]: Is this logging intended? If so, good to log a bit more context. Else remove it. ;-)

File: src/itest/kotlin/nl/lifely/zac/itest/client/KeycloakClient.kt [hunk 2]
```diff
@@ -22,16 +22,16 @@ object KeycloakClient {
     private const val ACCESS_TOKEN_ATTRIBUTE = "access_token"
     private const val REFRESH_TOKEN_ATTRIBUTE = "refresh_token"
 
-    fun authenticate() = okHttpClient.newCall(
+    fun authenticate(username: String = "testuser1", password: String = "testuser1") = okHttpClient.newCall(
```
Comment (edgarvonk) [inline src/itest/kotlin/nl/lifely/zac/itest/client/KeycloakClient.kt:25]: Better to use the existing `TEST_USER_1_ID` constant here for the username (and maybe rename it to `TEST_USER_1_USERNAME` come to think of it) and introduce a `TEST_USER_1_PASSWORD` constant in `ItestConfiguration` and use those here.

**Review by edgarvonk — CHANGES_REQUESTED** [review]

File: src/itest/kotlin/nl/lifely/zac/itest/ZakenRESTServiceTest.kt (hunk 1 above)
Comment (felixcicatt) [reply src/itest/kotlin/nl/lifely/zac/itest/ZakenRESTServiceTest.kt:437]: Good call, I'll remove it once I've got the test running on the build server

File: src/itest/kotlin/nl/lifely/zac/itest/client/KeycloakClient.kt (hunk 2 above)
Comment (felixcicatt) [reply src/itest/kotlin/nl/lifely/zac/itest/client/KeycloakClient.kt:25]: Absolutely right, will do

Comment (felixcicatt) [reply src/itest/kotlin/nl/lifely/zac/itest/client/KeycloakClient.kt]: Done

**Review by edgarvonk — APPROVED** [review]
Very nice

**Pull request merged**